src/redux/slices/productSlice.js

```diff
@@ -0,0 +1,151 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+export const getProducts = createAsyncThunk('products/getProducts', async (productParams) => {
+  const { data } = await axios.get('http://localhost:3000/products',
+    {
+      params: productParams
+    }
```

Comment:
K cần ghi cụ thể là productParams làm gì đâu anh. Vì file này là file `productSlice`, hàm này là hàm `getProducts`, thì chắc chắn 1 điều params này sẽ là cho product rồi
```suggestion
export const getProducts = createAsyncThunk('products/getProducts', async (params) => {
  const { data } = await axios.get('http://localhost:3000/products',
    {
      params
    }
```